packages/kbn-alerts-ui-shared/src/rule_form/rule_actions/rule_actions_alerts_filter.tsx

```diff
@@ -84,7 +84,18 @@ export const RuleActionsAlertsFilter = ({
   );
 
   const onFiltersUpdated = useCallback(
-    (filters: Filter[]) => updateQuery({ filters }),
+    (filters: Filter[]) => {
+      const updatedFilters = filters.map((filter) => {
+        const { $state, meta, ...rest } = filter;
+        return {
+          $state,
+          meta,
+          query: filter?.query ? { ...filter.query } : { ...rest },
```

Comment:
On an subsequent check I see that the filter is currently constructed in the wrong way and in general it doesn't conform to its type: `Filter`
```
type Filter = {
  $state?: {
    store: FilterStateStore;
  };
  meta: FilterMeta;
  query?: Record<string, any>;
};
```

The parameter `query` is not populated with the custom DSL, but the custom DSL always pollutes the rest of the object. I see this as a wrong implementation (not yours but the current unified search implementation) that needs to be fixed to make it work correctly in all cases.
IMHO what we are trying to push here is a workaround of a bigger problem, than an actual fix.

Reply:
> The parameter query is not populated with the custom DSL, but the custom DSL always pollutes the rest of the object.

Exactly!
Above changes had to be done because when filters are updated via `onFiltersUpdate`, the DSL filter value is not wrapped in a query which throws validation error in the rule's action as it is expected to be a `query` object.

https://github.com/user-attachments/assets/b6c89ca7-e0a7-43a4-b084-4a24f73c1a68


> IMHO what we are trying to push here is a workaround of a bigger problem, than an actual fix.

Agree, however as per my understanding not all plugins/ scenarios expects custom DSL value to be wrapped in a query object. Applying DSL filter to Alerts table works fine without query object. 
Hence I fixed it wherever it was needed. 
CC @cnasikas

Reply:
if you don't mind I'd like to see what could happen if I change your code like that: https://github.com/elastic/kibana/pull/195470/commits/9c8b5a302ed40e1c7740b2bd66622ec9fceb483b
let the CI run and see of this change trigger any error in the use of the unified search. This definitely work with your use case, but I'd like to check if this causes problem in the surrounding

Reply:
[added](https://github.com/elastic/kibana/pull/193623/commits/b663e2b92a76f76f2b9ff4e8a21c1b9ad64f811e)